py/argcheck.c

```diff
@@ -139,7 +139,7 @@ void mp_arg_parse_all_kw_array(mp_uint_t n_pos, mp_uint_t n_kw, const mp_obj_t *
     mp_arg_parse_all(n_pos, args, &kw_args, n_allowed, allowed, out_vals);
 }
 
-#if MICROPY_ERROR_REPORTING == MICROPY_ERROR_REPORTING_TERSE
+#if MICROPY_ERROR_REPORTING == MICROPY_ERROR_REPORTING_TERSE || _MSC_VER
```

Comment:
It has to be a gcc optimization. In debug mode cl compiles `if (MICROPY_ERROR_REPORTING == MICROPY_ERROR_REPORTING_TERSE) into `xor eax,eax` followed by a jump which is always skipped of course. In release mode it gets rid of everything, yet still wants the function definition to be available.